dashboard/test/controllers/api/v1/pd/workshop_survey_foorm_report_controller_test.rb

```diff
@@ -147,17 +155,78 @@ class WorkshopSurveyFoormReportControllerTest < ::ActionController::TestCase
-      create_list :csf_intro_post_facilitator_workshop_submission,
-        low_count,
-        :answers_low,
-        pd_workshop_id: csf_workshop.id,
-        facilitator_id: facilitator_id
-      create_list :csf_intro_post_facilitator_workshop_submission,
-        high_count,
-        :answers_high,
-        pd_workshop_id: csf_workshop.id,
-        facilitator_id: facilitator_id
+    test 'return unauthorized for unauthorized users' do
+      generic_teacher = create :teacher
+      other_facilitator = create :facilitator
+      program_manager = create :program_manager
+      workshop_organizer = create :workshop_organizer
+      csf_workshop = create :csf_workshop,
+        started_at:  Time.now.utc - 1.day,
+        ended_at: Time.now.utc - 1.hour
+
+      this_facilitator = csf_workshop.facilitators[0]
+
+      expected_authorization = [
+        {user: this_facilitator, expected_response: :success, type: "facilitator for this workshop"},
+        {user: program_manager, expected_response: :success, type: "program manager"},
+        {user: workshop_organizer, expected_response: :success, type: "workshop organizer"},
+        {user: @workshop_admin, expected_response: :success, type: "workshop admin"},
+        {user: generic_teacher, expected_response: :unauthorized, type: "teacher"},
+        {user: other_facilitator, expected_response: :unauthorized, type: "other facilitator"}
+      ]
+
+      expected_authorization.each do |data|
+        sign_in data[:user]
+        get :generic_survey_report, params: {workshop_id: csf_workshop.id}
+        assert_response data[:expected_response], "#{data[:type]} had an unexpected result"
+        sign_out data[:user]
+      end
+    end
+
+    test 'filters facilitator data if facilitator is signed in' do
+      csf_workshop = create :csf_workshop,
+        started_at:  Time.now.utc - 1.day,
+        ended_at: Time.now.utc - 1.hour,
+        num_facilitators: 2
+      facilitator_1 = csf_workshop.facilitators[0]
+      facilitator_2 = csf_workshop.facilitators[1]
+      create_surveys_for_csf_workshop(csf_workshop, csf_workshop.facilitators.pluck(:id), 5, 2)
+
+      sign_in facilitator_1
+      get :generic_survey_report, params: {workshop_id: csf_workshop.id}
+      assert_response :success
+
+      response = JSON.parse(@response.body, symbolize_names: true)
+
+      facilitator_1_id = facilitator_1.id.to_s.to_sym
+      facilitator_2_id = facilitator_2.id.to_s.to_sym
+
+      assert_equal 7, response[:this_workshop][:Overall][:facilitator][:response_count][facilitator_1_id]
+
+      overall_facilitator = response[:this_workshop][:Overall][:facilitator]['surveys/pd/workshop_csf_intro_post.0'.to_sym]
+      facilitator_effectiveness = overall_facilitator[:facilitator_effectiveness]
+      assert_not_nil facilitator_effectiveness[facilitator_1_id][:on_track]
+      assert_nil facilitator_effectiveness[facilitator_2_id]
```

Comment:
As discussed, might be good to emphasise these two `assert_nil` calls (with a comment) because they are very important moments where we ensure that a facilitator doesn't see another facilitator's results.